src/cpuid/src/template/intel/configurable_template.rs

```diff
@@ -0,0 +1,234 @@
+                                    cpu_feature_mapping.feature_mapping.bit_index,
+                                    cpu_feature_mapping.config.is_enabled,
+                                );
+                            }
+                            Register::ECX => {
+                                entry.ecx.write_bit(
+                                    cpu_feature_mapping.feature_mapping.bit_index,
+                                    cpu_feature_mapping.config.is_enabled,
+                                );
+                            }
+                            Register::EDX => {
+                                entry.edx.write_bit(
+                                    cpu_feature_mapping.feature_mapping.bit_index,
+                                    cpu_feature_mapping.config.is_enabled,
+                                );
+                            }
+                            Register::MSR { .. } => {}
+                        }
+                    }
+                }
+            }
+        }
+    };
+}
+
+fn build_single_msr_configuration(
+    msr_pointer: &Register,
+    msr_enabled_features: Option<&Vec<&CpuConfigurationInstruction>>,
+) -> Option<kvm_msr_entry> {
+    match msr_pointer {
+        Register::MSR { addr } => match msr_enabled_features {
+            None => None,
+            Some(msr_features) => {
+                let mut capabilities: u32 = 0;
+                for &feature in msr_features {
+                    warn!(
+                        "MSR feature config - [{}] = {}",
+                        feature.config.name, feature.config.is_enabled
+                    );
+                    if feature.config.is_enabled {
+                        capabilities = capabilities | feature.feature_mapping.bit_index;
+                    }
+                }
+
+                Some(kvm_msr_entry {
+                    index: *addr,
+                    data: capabilities as u64,
+                    ..Default::default()
+                })
+            }
+        },
+        _ => None,
+    }
+}
+
+/// Type to help demarcate how a given CPU feature must be configured
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum CpuRegisterFeatureType {
+    /// CPU feature to be configured via normal CPUID interface as per x86 arch
+    CPUID,
+    /// CPU feature that requires configuration via a model specific register
+    MSR,
+}
+
+/// Encapsulates information necessary to toggle an arbitrary CPU feature
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub struct CpuFeatureArchMapping {
+    leaf: Option<u32>,
+    bit_index: u32,
+    register: Register,
+    feature_type: CpuRegisterFeatureType,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+struct CpuConfigurationInstruction {
+    feature_mapping: CpuFeatureArchMapping,
+    config: CpuConfigurationAttribute,
+}
+
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Hash)]
+enum Register {
+    EAX,
+    EBX,
+    ECX,
+    EDX,
+    MSR { addr: u32 },
```

Comment:
You could use a [newtype](https://rust-unofficial.github.io/patterns/patterns/behavioural/newtype.html) here.
```suggestion
enum Register {
    EAX,
    EBX,
    ECX,
    EDX,
    MSR(u32),
```

Reply:
I do find the value being named does help quite a bit, readability wise as well as syntactically.